### PR TITLE
3387 fix threshold value issue in AsDiscrete

### LINKED ISSUE #3387: `AsDiscreted` bug when `threshold=0.0`
Reproduce the bug:

```
from monai.transforms import AsDiscreted
a = AsDiscreted(keys="pred", threshold=0.0)
```

Screen Shot 2021-11-23 at 10 56 39 PM

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #3387 .

### Description
This PR fixed the value=0 issue in AsDiscrete transform.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/post/dictionary.py [hunk 1]
```diff
@@ -175,13 +175,13 @@ def __init__(
...
             warnings.warn("`to_onehot=True/False` is deprecated, please use `to_onehot=num_classes` instead.")
             num_classes = ensure_tuple_rep(num_classes, len(self.keys))
             self.to_onehot = tuple(val if flag else None for flag, val in zip(self.to_onehot, num_classes))
 
         self.threshold = ensure_tuple_rep(threshold, len(self.keys))
-        if True in self.threshold or False in self.threshold:  # backward compatibility
+        if any([isinstance(i, bool) for i in self.threshold]):  # backward compatibility
             warnings.warn("`threshold_values=True/False` is deprecated, please use `threshold=value` instead.")
             logit_thresh = ensure_tuple_rep(logit_thresh, len(self.keys))
             self.threshold = tuple(val if flag else None for flag, val in zip(self.threshold, logit_thresh))
```
Comment (wyli) [inline monai/transforms/post/dictionary.py:187]: I think this `flag` check will also need update

Comment (Nic-Ma) [reply monai/transforms/post/dictionary.py:187]: Sure, updated in the latest commit.

Thanks.

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me.

**Pull request merged**